controllers/users.js

```diff
@@ -602,6 +604,51 @@ const setInDiscordScript = async (req, res) => {
   }
 };
 
+/**
+ * Collects all Users with ONBOARDING state and are present in discord server for more than 31 days
+ *
+ * @param req {Object} - Express request object
+ * @param res {Object} - Express response object
+ */
+
+const getUsersWithOnboardingState = async (req, res) => {
+  const minimumPresentDaysInServer = 31;
+  try {
+    const { allUserStatus } = await userStatusModel.getAllUserStatus(req.query);
+
+    const allUsersWithOnboardingState = filterUsersWithOnboardingState(allUserStatus);
+
+    if (!allUsersWithOnboardingState.length) {
+      return res.boom.notFound("No users exist with an 'ONBOARDING' state");
+    }
+
+    const updatedOnboardingUsersWithDate = [];
+    for (const user of allUsersWithOnboardingState) {
+      const result = await userQuery.fetchUser({ userId: user.userId });
+      if (result.user.discordJoinedAt) {
+        const userDiscordJoinedDate = new Date(result.user.discordJoinedAt);
+        const currentDate = new Date();
+        const timeDifferenceInMilliseconds = currentDate.getTime() - userDiscordJoinedDate.getTime();
+        const currentAndUserJoinedDateDifference = Math.floor(timeDifferenceInMilliseconds / (1000 * 60 * 60 * 24));
+        if (currentAndUserJoinedDateDifference > minimumPresentDaysInServer) {
+          updatedOnboardingUsersWithDate.push(result.user);
+        }
+      }
+    }
+
+    return res.json({
+      message: updatedOnboardingUsersWithDate.length
+        ? `All Users with ONBOARDING state of more than ${minimumPresentDaysInServer} days found successfully`
+        : `No users exist with ONBOARDING state of more than ${minimumPresentDaysInServer} days, or the user has not been verified`,
+      totalUsers: updatedOnboardingUsersWithDate.length,
+      allUser: updatedOnboardingUsersWithDate,
+    });
+  } catch (err) {
+    logger.error(`Error while fetching all the User: ${err}`);
+    return res.boom.badImplementation(INTERNAL_SERVER_ERROR);
```

Comment:
500 error will come for any line in the given function. We cannot assume on what condition/basis/line of code/ the error will give 500. It can happen if Firestore is down or some other third-party dependent service is down.
Even if we test for firestore but what if something else caused the issue and the test passes for it?
Ideally, 500 errors are not in our control.
What do you think, let me know.